tensorflow/core/kernels/mkl_qmatmul_op.cc

```diff
@@ -0,0 +1,875 @@
+ public:
+  virtual ~MklDnnQuantizedMatMulReluOp() {}
+
+  explicit MklDnnQuantizedMatMulReluOp(OpKernelConstruction* context)
+      : MklDnnQuantizedMatMulOp<Device, Tinput, Tweight, Tbias, Toutput>(
+            context) {}
+
+ protected:
+  void ExtendMklDnnMatMulFwdParams(OpKernelContext* context,
+                                   MklDnnMatMulFwdParams& params) override {
+    MklDnnQuantizedMatMulOp<Device, quint8, qint8, Tbias,
+                            Toutput>::ExtendMklDnnMatMulFwdParams(context,
+                                                                  params);
+    params.post_op_params.push_back({"relu", {1.0, 0.0, 0.0}});
+  }
+};
+
+// Register NoOp kernel for QuantizedMatMulWithBias to get a python interface.
+// This kernel will be replaced by an MKL kernel during graph
+// optimization pass.
+REGISTER_KERNEL_BUILDER(Name("QuantizedMatMulWithBias")
+                            .Device(DEVICE_CPU)
+                            .TypeConstraint<quint8>("T1")
+                            .TypeConstraint<qint8>("T2")
+                            .TypeConstraint<qint32>("Toutput"),
+                        NoOp);
+
+REGISTER_KERNEL_BUILDER(
+    Name("_MklQuantizedMatMulWithBias")
+        .Device(DEVICE_CPU)
+        .TypeConstraint<quint8>("T1")
+        .TypeConstraint<qint8>("T2")
+        .TypeConstraint<float>("Tbias")
+        .TypeConstraint<qint32>("Toutput")
+        .Label(mkl_op_registry::kMklQuantizedOpLabel),
+    MklDnnQuantizedMatMulOp<CPUDevice, quint8, qint8, float, qint32>);
+REGISTER_KERNEL_BUILDER(
+    Name("_MklQuantizedMatMulWithBias")
+        .Device(DEVICE_CPU)
+        .TypeConstraint<quint8>("T1")
+        .TypeConstraint<qint8>("T2")
+        .TypeConstraint<qint32>("Tbias")
+        .TypeConstraint<qint32>("Toutput")
+        .Label(mkl_op_registry::kMklQuantizedOpLabel),
+    MklDnnQuantizedMatMulOp<CPUDevice, quint8, qint8, qint32, qint32>);
+
+// Register NoOp kernel for QuantizedMatMulWithBiasAndRelu to get a python
+// interface. This kernel will be replaced by an MKL kernel during
+// graph-optimization pass.
+REGISTER_KERNEL_BUILDER(Name("QuantizedMatMulWithBiasAndRelu")
+                            .Device(DEVICE_CPU)
+                            .TypeConstraint<quint8>("T1")
+                            .TypeConstraint<qint8>("T2")
+                            .TypeConstraint<qint32>("Toutput"),
+                        NoOp);
+
+// Register NoOp kernel for QuantizedIPWithBiasAndReluAndRequantize
+// to get a python interface. This kernel will be replaced by an MKL kernel
+// during graph-optimization pass.
+REGISTER_KERNEL_BUILDER(Name("QuantizedMatMulWithBiasAndReluAndRequantize")
+                            .Device(DEVICE_CPU)
+                            .TypeConstraint<quint8>("T1")
+                            .TypeConstraint<qint8>("T2")
+                            .TypeConstraint<qint32>("Tbias")
```

Comment:
You can specify multiple types with this list form:
```suggestion
                            .TypeConstraint("Tbias", {DT_QINT32, DT_FLOAT})
```
See
https://github.com/tensorflow/tensorflow/blob/273b97387054474536bbe441301255d0d23d4ab8/tensorflow/core/framework/op_kernel.h#L1328-L1332